airflow/models/abstractoperator.py

```diff
@@ -654,6 +654,7 @@ def expand_mapped_task(self, run_id: str, *, session: Session) -> tuple[Sequence
                     unmapped_ti.map_index = 0
                     self.log.debug("Updated in place to become %s", unmapped_ti)
                     all_expanded_tis.append(unmapped_ti)
+                    task_instance_mutation_hook(unmapped_ti)
```

Comment:
Can you please add a comment "why" you call it here so that somebody in the future knows why?

Reply:
Good idea added comment and added unit test.